File: leetcode/remove_element.py

from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        index = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[index] = nums[i]
                index += 1
        return index
    
if __name__ == "__main__":
    s = Solution()
    nums = [3, 2, 2, 3, 4, 5, 7]
    val = 3
    print(s.removeElement(nums, val))

File: leetcode/remove_element.cpp
using namespace std;
#include <iostream>
#include <vector>

class Solution {
public: 
    int removeElement(vector<int>&nums, int val) {
        int index = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val) {
                nums[index] = nums[i];
                index += 1;
            }
        }
        return index;
    }
};

int main() {
    Solution s;
    vector<int> nums = {3, 2, 2, 3, 5, 7};
    int val = 3;
    cout << s.removeElement(nums, val) << endl;
}